[Merged by Bors] - Minor logging improvements

lrettig opened the pull request:
## Motivation
This PR contains a hodgepodge of small improvements to logging that have been running in the past few testnets, useful for debugging.

## Changes
- More contextual logging
- Adds framework for simple logging of memory stats
- More verbose and explicit logging
- Some minor cleanup of logs and comments

NO logic changes.

## Test Plan
N/A - no logic changes

## TODO
<!-- This section should be removed when all items are complete -->
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes
<!-- Please uncheck these items as applicable to make DevOps aware of changes that may affect releases -->
- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [ ] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

lrettig committed:
Minor cleanup of logs and comments

lrettig committed:
Tweak logging for testnet debugging

lrettig committed:
Tweak logging for debug purposes

To help with debugging of
https://github.com/spacemeshos/go-spacemesh/issues/2405

lrettig committed:
Properly handle errors in meshdb

lrettig committed:
Logging improvements

for debugging

lrettig committed:
Logging

lrettig committed:
More context and logging improvements

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig committed:
Bugfix

lrettig committed:
Logging improvement for state, p2p

For testnet debugging

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig committed:
Improve logging/context for debugging

For debugging late hare messages. See
https://github.com/spacemeshos/go-spacemesh/issues/2386#issuecomment-840174414

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig committed:
Add simple, periodic logging of memory usage

lrettig committed:
Add logs for ATX publishing

Small bugfix

lrettig committed:
Minor logging improvements

lrettig committed:
Make logs searchable by module name

Much simpler than the previous fix

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig committed:
Improve logging for p2p debugging

Continue debugging #2386

lrettig committed:
Make logs searchable by module: take three

lrettig committed:
Fix log tests

noamnelke reviewed (approved):
TIL about error wrapping in Go 🎓

lrettig committed:
Revert "Make logs searchable by module name"

This reverts commit 5dab96717a4905f5e9d66faeb2ec6fc4150acfd7.

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig committed:
Resolve merge conflict

lrettig committed:
Minor cleanup

Remove some unused code

lrettig committed:
Bugfix in meshdb

Update layer with block in writeBlock

lrettig committed:
Improve errmsg

lrettig committed:
Revert "Add simple, periodic logging of memory usage"

This reverts commit 057c5d1d2413f8fa9c901f5ee481fd18b5c07407.

lrettig commented:
I removed the memlogger stuff since I'm not certain we want that in the develop branch. I moved it to #2451. @noamnelke could you take one more look please, esp. at the final iteration of the logging changes?

lrettig committed:
Increase threshold for queue backlog warnings

noamnelke commented on blocks/blockoracle.go:
```diff
@@ -213,10 +213,10 @@ func getNumberOfEligibleBlocks(weight, totalWeight uint64, committeeSize uint32,
-func (bo *Oracle) getATXIDForEpoch(targetEpoch types.EpochID) (types.ATXID, error) {
-	latestATXID, err := bo.atxDB.GetNodeAtxIDForEpoch(bo.nodeID, targetEpoch)
+func (bo *Oracle) getATXIDForEpoch(epoch types.EpochID) (types.ATXID, error) {
+	latestATXID, err := bo.atxDB.GetNodeAtxIDForEpoch(bo.nodeID, epoch)
```
ATXs are associated with at two epochs: a target epoch and a publication epoch. This was disambiguating and could become more confusing, roll back?

noamnelke commented on log/zap.go:
```diff
@@ -185,30 +192,30 @@ var Nop = zap.WrapCore(func(zapcore.Core) zapcore.Core {
 // Info prints message with fields
 func (fl FieldLogger) Info(msg string, fields ...LoggableField) {
-	fl.l.Info(msg, unpack(fields)...)
+	fl.l.Info(msg, unpack(append(fields, String("name", fl.name)))...)
```
isn't it better to add a field to the logger than special-case it like this?

lrettig replied:
The problem is, once you add a field, you can't remove it or change it. If we add `name` as a field, then if a child logger is created, it will get two `name` fields, e.g., `{name: "p2p", name: "p2p.TcpNet"}`. This logic is smarter, see the code at: https://github.com/spacemeshos/go-spacemesh/pull/2427/files/c1c336a3c685bef597d5dcab0f83bd729dc13cfa#diff-6e36b5b54c2b1b501474b0cb1be773f051a3710ae1754f16ce1e479b4577c46bR148-R157

lrettig replied:
I added an explanatory comment to this effect

noamnelke replied:
Oh, I assumed that fields are entries in a map-like object, that can be overwritten. I just followed the code and realized this is not the case. Counter-intuitive and less useful, if you ask me, but 🤷🏻‍♂️.

noamnelke commented on p2p/message.go:
```diff
@@ -61,8 +63,13 @@ func (pm gossipProtocolMessage) ValidationCompletedChan() chan service.MessageVa
-func (pm gossipProtocolMessage) ReportValidation(protocol string) {
+func (pm gossipProtocolMessage) ReportValidation(ctx context.Context, protocol string) {
 	if pm.validationChan != nil {
+		log.AppLog.WithContext(ctx).With().Info("reporting valid gossip message",
```
This will result in a message with info level being logged by every node for every gossip message, it could massively increase the load on our logging infra. While I can see why it would be useful sometimes, maybe we can find a way to be more selective?

lrettig replied:
you're right. I want Info level for the testnet since this is helpful in debugging active issues, but it should be Debug here. I'll fix it.

lrettig committed:
Pass context to IsIdentityActiveOnConsensusView

lrettig committed:
Log tweaks

lrettig committed:
Make sure gossip msghash is logged

lrettig committed:
Improve logging

For hare-related testnet debugging

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig committed:
Add explanatory comment

noamnelke reviewed (changes requested):
You resolved [this comment](https://github.com/spacemeshos/go-spacemesh/pull/2427#discussion_r653537637) and [this one](https://github.com/spacemeshos/go-spacemesh/pull/2427#discussion_r654895041) although they were not addressed.

Otherwise I think it's good to go.

lrettig committed:
PR review feedback

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig commented:
> You resolved [this comment](https://github.com/spacemeshos/go-spacemesh/pull/2427#discussion_r653537637) and [this one](https://github.com/spacemeshos/go-spacemesh/pull/2427#discussion_r654895041) although they were not addressed.

Sorry about that, I must've gotten mixed up when resolving a merge conflict or something. All taken care of.

Please approve if it looks good. Thanks!

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig committed:
Lower test threshold

In hopes of passing p2p systest

lrettig committed:
Lower test threshold

In hopes of passing p2p systest

lrettig committed:
Lower test threshold

In hopes of passing p2p systest

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig committed:
Revert "Lower test threshold"

This reverts commit 662aca4fe27394a1adb66081a1a49a1d6fe41b3d.

lrettig committed:
Revert "Lower test threshold"

This reverts commit 662aca4fe27394a1adb66081a1a49a1d6fe41b3d.

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig commented:
bors merge

lrettig commented:
bors merge

>

lrettig commented:
bors merge

lrettig committed:
Merge branch 'develop' into testnet-debugging

lrettig commented:
bors merge

lrettig commented:
bors merge

On Thu, Jul 1, 2021 at 23:20 bors[bot] ***@***.***> wrote:

> Build failed:
>
>    - ci-stage2
>    <https://github.com/spacemeshos/go-spacemesh/runs/2968804186>
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spacemeshos/go-spacemesh/pull/2427#issuecomment-872683440>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZJWNBKOCTU7PX2ODCANQTTVUWAHANCNFSM45NVQV3A>
> .
>

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig committed:
Revert "Lower test threshold"

This reverts commit e67da6d8f687e6f63d48ddbd4e0e406328ccfa62.

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig committed:
Lower p2p test params

lrettig commented:
bors merge

lrettig commented:
bors merge

lrettig commented:
bors merge
